src/tests/suite/daos_test.h

```diff
@@ -439,7 +439,17 @@ test_rmdir(const char *path, bool force)
 		if (!strcmp(ent->d_name, ".") || !strcmp(ent->d_name, ".."))
 			continue;   /* skips the dots */
 
-		sprintf(fullpath, "%s/%s", path, ent->d_name);
+		rc = snprintf(fullpath, PATH_MAX, "%s/%s", path, ent->d_name);
+		if (rc < 0 || rc >= PATH_MAX) {
+			if (rc < 0)
+				rc = -DER_INVAL;	/* invalid */
+			if (rc >= PATH_MAX)
+				rc = -DER_NOMEM;	/* overflow */
+			D_FREE(fullpath);
```

Comment:
If you initialize freepath = NULL, you can safely call D_FREE in one place after the out label.   According to the manpage for free, null is a supported noop.  The macro may actually check for NULL as well to avoid calling free.